File: app-kotlin/src/main/java/fudi/freddy/mokitosample/contact_view/ContactContract.java
package fudi.freddy.mokitosample.contact_view;

import fudi.freddy.mokitosample.model.Contact;

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */

public interface ContactContract {

    interface ContactView{
        void clearText();
        String loadInput1();
        String loadInput2();
        String loadInput3();
        String loadInput4();
        String loadInput5();
        Contact getContactView();
        void display(String message);
        void display(Contact contact);

    }

    interface UserActions{
        void saveContact();
        void loadContact(String mail);
    }
}

File: app-kotlin/src/main/java/fudi/freddy/mokitosample/contact_view/ContactContract.kt
package fudi.freddy.mokitosample.contact_view

import fudi.freddy.mokitosample.model.Contact

/**
 * @version 1.0
 * @autor Romell Domínguez
 * @date 8/13/17
 */
interface ContactContract {
    interface ContactView {
        fun clearText()
        fun loadInput1(): String
        fun loadInput2(): String
        fun loadInput3(): String
        fun loadInput4(): String
        fun loadInput5(): String
        val contact: Contact
        fun display(message: String?)
        fun display(contact: Contact?)
    }

    interface UserActions {
        fun saveContact()
        fun loadContact(mail: String?)
    }
}